src/SFA.DAS.EAS.Web/Orchestrators/CommitmentStatusCalculator.cs

```diff
@@ -53,5 +56,32 @@ private RequestStatus GetEmployerOnlyStatus(LastAction lastAction, bool hasAppre
 
             return RequestStatus.None;
         }
+
+        public RequestStatus GetTransferStatus(EditStatus edit, TransferApprovalStatus transferApproval)
+        {
+            const string invalidStateExceptionMessagePrefix = "Transfer funder commitment in invalid state: ";
+
+            if (edit >= EditStatus.Neither)
+                throw new Exception("Unexpected EditStatus");
+
+            switch (transferApproval)
+            {
+                case TransferApprovalStatus.Pending:
+                    return edit == EditStatus.Both ? RequestStatus.WithSender : RequestStatus.NewRequest;
+
+                case TransferApprovalStatus.Approved:
+                    if (edit != EditStatus.Both)
+                        throw new InvalidStateException($"{invalidStateExceptionMessagePrefix}If approved by sender, must be approved by receiver and provider");
+                    return RequestStatus.None;
+
+                case TransferApprovalStatus.Rejected:
+                    if (edit != EditStatus.EmployerOnly)
+                        throw new InvalidStateException($"{invalidStateExceptionMessagePrefix}If just rejected by sender, must be with receiver");
+                    return RequestStatus.WithSender;
```

Comment:
It might be best to have two new RequestStatus enum values - one for Pending, the other for Rejected, and then roll those two into the single bingo box. Reason being, when we view the cohort, we will want to show the actual status.

Reply:
Implemented in DPP-1229 View transfer-funded cohorts